ros/angel_system_nodes/angel_system_nodes/audio_player.py

```diff
@@ -0,0 +1,85 @@
+import copy
+import threading
+import time
+
+import simpleaudio as sa
+
+import rclpy
+from rclpy.node import Node
+from angel_msgs.msg import HeadsetAudioData
+
+
+class AudioPlayer(Node):
+    def __init__(self):
+        super().__init__(self.__class__.__name__)
+
+        self.declare_parameter("audio_topic", "HeadsetAudioData")
+
+        self._audio_topic = self.get_parameter("audio_topic").get_parameter_value().string_value
+
+        self.subscription = self.create_subscription(
+            HeadsetAudioData,
+            self._audio_topic,
+            self.listener_callback,
+            1)
+
+        self.audio_stream = bytearray()
+        self.audio_player_lock = threading.RLock()
+        self.t = threading.Thread()
+
+        self.audio_stream_duration = 0.0
+        self.playback_duration = 1.0
+
+    def listener_callback(self, msg):
+        """
+        Add the audio to the cumulative audio stream and spawns a new playback
+        thread if the cumulative recording has reached more than a second and
+        there is no ongoing playback.
+        """
+        self.audio_stream.extend(msg.data)
```

Comment:
I'm not sure how ROS handles threading this, but this listener might want to use a lock to become a critical section so that message interleaving doesn't occur, probably checking that the current message is temporally after our last message.

Reply:
I added some code to check that the messages are sequential and a lock around the audio_stream. Let me know if you were thinking something else.